rust/src/applayer.rs

```diff
@@ -386,6 +386,17 @@ macro_rules! build_slice {
     ($buf:ident, $len:expr) => ( std::slice::from_raw_parts($buf, $len) );
 }
 
+#[macro_export]
+macro_rules! build_slice_null {
```

Comment:
seems to belong in it's own commit

Also, should this be used in the `as_slice` thing as well?

Reply:
It is in its own commit. What do you mean ?

> Also, should this be used in the as_slice thing as well?

I think I tried and did not get it to compile with magic macro...
But then, we would also have to refactor all usages of from_raw_parts to use either macro...

Reply:
`build_slice_null` addition in its own commit, explaining it, instead of making it part of some http2 commit